### PR TITLE
Sass 286

### PR DESCRIPTION
**Pull request by dnyuar**

### CODE REVIEW

**Comment by robert-webb-hmrc** [comment]
Can we get a rebase on this please?

File: app/connectors/httpparsers/DividendsSubmissionHttpParser.scala [hunk 1]
```diff
@@ -16,28 +16,54 @@
...
+    Some(s"[DividendsSubmissionHttpParser][read] Received ${response.status} from API. Body:${response.body}")
+  }
+
+  private def handleApiError(response: HttpResponse, statusOverride: Option[Int] = None): DividendsSubmissionsResponse = {
+    val status = statusOverride.getOrElse(response.status)
+
+    try {
+      response.json.validate[ApiErrorBodyModel].fold[DividendsSubmissionsResponse](
+        jsonErrors => {
+          pagerDutyLog(UNEXPECTED_RESPONSE_FROM_API, Some(s"[DividendsSubmissionHttpParser][read] Unexpected Json from DES."))
```
Comment (BenCosfordHMRC) [inline app/connectors/httpparsers/DividendsSubmissionHttpParser.scala:59]: change from DES to from API

File: app/models/ApiErrorBodyModel.scala [hunk 2]
```diff
@@ -12,14 +12,15 @@
...
+package models
 
-@this(layout: Layout)
+import play.api.libs.json.{Json, OFormat}
 
-@(pageTitle: String, heading: String, message: String)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
-@layout(pageTitle = pageTitle) {
- <h1 class="govuk-heading-xl">@{Text(heading).asHtml}</h1>
- <p class="govuk-body">@{Text(message).asHtml}</p>
+case class ApiErrorBodyModel(code: String, description: String)
```
Comment (BenCosfordHMRC) [inline app/models/ApiErrorBodyModel.scala:21]: Swap description to reason as per DES specs.

File: it/connectors/InterestSubmissionConnectorISpec.scala [hunk 3]
```diff
@@ -55,31 +56,58 @@ class InterestSubmissionConnectorISpec extends IntegrationTest {
...
     }
 
     "return an internal server error response" when {
 
       "one is retrieved from the endpoint" in {
-        stubPost(s"/income-tax-interest/income-tax/nino/$nino/sources\\?taxYear=$taxYear&mtditid=$mtditid", INTERNAL_SERVER_ERROR, "{}")
+
+        val responseBody = Json.obj(
+          "code" -> "INTERNAL_SERVER_ERROR",
+          "description" -> "there has been an error downstream"
```
Comment (BenCosfordHMRC) [inline it/connectors/InterestSubmissionConnectorISpec.scala:70]: Swap description to reason.

File: app/connectors/httpparsers/DividendsSubmissionHttpParser.scala (hunk 1 above)
Comment (dnyuar) [reply app/connectors/httpparsers/DividendsSubmissionHttpParser.scala:59]: changed

**Comment by dnyuar** [comment]
I've pushed up the changes to this branch

**Pull request merged**